ci_scripts/test_sync_target_db.py

```diff
@@ -1,147 +1,123 @@
+                "IC RSS (Canada)",
+                "ICASA (South Africa)",
+                "KCC (South Korea)",
+                "MIC (Japan)",
+                "NCC (Taiwan)",
+                "RoHS (Europe)",
+            ],
+            "communication": ["Bluetooth & BLE"],
+            "interface_firmware": ["DAPLink", "J-Link"],
+            "target_core": ["Cortex-M4"],
+            "mbed_studio_support": ["Build and run"],
+        },
+        "board_type": "MTB_UBLOX_NINA_B1",
+        "flash_size": 512,
+        "name": "u-blox NINA-B1",
+        "product_code": "0455",
+        "ram_size": 64,
+        "target_type": "module",
+        "hidden": False,
+        "device_name": "nRF52832_xxAA",
+    },
+}
 
-ONLINE_TARGET_DATABASE = f"[{TARGET_1}, {TARGET_2}]"
+TARGET_2 = {
+    "type": "target",
+    "id": "2",
+    "attributes": {
+        "features": {
+            "mbed_enabled": ["Advanced"],
+            "mbed_os_support": [
+                "Mbed OS 5.10",
+                "Mbed OS 5.11",
+                "Mbed OS 5.12",
+                "Mbed OS 5.13",
+                "Mbed OS 5.14",
+                "Mbed OS 5.15",
+                "Mbed OS 5.8",
+                "Mbed OS 5.9",
+            ],
+            "antenna": ["Connector"],
+            "certification": ["AS/NZS (Australia and New Zealand)", "CE (Europe)", "FCC/CFR (USA)", "IC RSS (Canada)"],
+            "communication": ["LoRa"],
+            "interface_firmware": ["DAPLink"],
+            "target_core": ["Cortex-M3"],
+            "mbed_studio_support": ["Build and run"],
+        },
+        "board_type": "MTB_MTS_XDOT",
+        "flash_size": 256,
+        "name": "Multitech xDOT",
+        "product_code": "0453",
+        "ram_size": 64,
+        "target_type": "module",
+        "hidden": False,
+        "device_name": "STM32L151CC",
+    },
+}
 
-OFFLINE_TARGET_DATABASE = f"[{TARGET_1}]"
 
-EXPECTED_DIFF = f"[{TARGET_2}]"
+def _make_mbed_targets_for_diff(targets_a, targets_b):
+    return MbedTargets(targets_a), MbedTargets(targets_b)
 
 
 class TestSyncTargetDB(TestCase):
-    def test_diff_objects_from_json_returns_diffs(self):
-        with mock.patch("ci_scripts.sync_target_database.MbedTargets") as mock_online_targets:
-            mock_online_targets.json_dump.return_value = ONLINE_TARGET_DATABASE
-            diffs = sync_target_database.diff_objects_from_json_str(
-                mock_online_targets.json_dump(), OFFLINE_TARGET_DATABASE
-            )
-            self.assertEqual(diffs, json.loads(EXPECTED_DIFF), "Actual diff and expected diff should match")
+    def test_get_boards_added_or_removed_detects_added(self):
+        mock_online_targets, mock_offline_targets = _make_mbed_targets_for_diff([TARGET_1, TARGET_2], [TARGET_1])
+        added, removed = sync_target_database.get_boards_added_or_removed(mock_offline_targets, mock_online_targets)
+        self.assertEqual(len(added), 1, "One new board to be added to offline db.")
```

Comment:
NIT: I'm a bit conflicted about those custom error messages. I.e.:
1. `One new board to be added to offline db.` -> `Expected exactly one new board to be added to offline db.`
2. This one in particular: `No boards removed from offline db.` -> `Expected no boards to be removed from offline db.`
3. `Should return an empty targets container when no targets added` -> `Returned a non-empty target container when no targets added`.
Etc.

NIT^2: "Should" sounds like asking, yet in my mind computer must obey what I tell it to do. It's not a _maybe_, it's an _expectation_. Not only "should" takes up space, but is meaningless in the grand scheme of things. "Should return" vs "returns"; "should to something" vs "does something" etc. 😉

Reply:
I will remove "should" - although I'm not convinced adding the word 'expected' to the other messages improves them. This is the point where I say we should be using pytest anyway but I'm not going to go there :)